server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -114,14 +114,41 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
     final ObjectMapper objectMapper = isSmile ? smileMapper : jsonMapper;
     request.setAttribute(OBJECTMAPPER_ATTRIBUTE, objectMapper);
 
-    String host = hostFinder.getDefaultHost();
-    request.setAttribute(HOST_ATTRIBUTE, host);
-
-    boolean isQuery = request.getMethod().equals(HttpMethod.POST.asString()) &&
-                      request.getRequestURI().startsWith("/druid/v2");
-
-    // queries only exist for POST
-    if (isQuery) {
+    final String defaultHost = hostFinder.getDefaultHost();
+    request.setAttribute(HOST_ATTRIBUTE, defaultHost);
+
+    final boolean isQueryEndpoint = request.getRequestURI().startsWith("/druid/v2");
+
+    if (isQueryEndpoint && request.getMethod().equals(HttpMethod.DELETE.asString())) {
+      // query cancellation request
+      for (final String host : hostFinder.getAllHosts()) {
```

Comment:
if there's an error in submitting to one of the hosts we probably want to try the others.